FX.Core.OCR/Barcodes/Implementations/BarcodeDecoder.cs

```diff
@@ -0,0 +1,163 @@
+﻿using FX.Core.OCR.Barcodes.Models;
+using FX.Core.OCR.Barcodes.Settings;
+using FX.Core.OCR.Barcodes.Utils;
+using FX.Core.OCR.Barcodes.Transformer.Implementations;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using ZXing;
+using ZXing.Common;
+using ZXing.Windows.Compatibility;
+using FX.Core.OCR.Barcodes.Abstract;
+
+namespace FX.Core.OCR.Barcodes.Implementations
+{
+    class BarcodeDecoder : IBarcodeDecoder
+    {
+        private readonly BarcodeConfig _config;
+        private readonly ZXing.BarcodeReader barcodeReader;
+        private long _timeStart;
+
+        private readonly IList<DecodeResult> results;
+        private DecodeStatistics statistics;
+
+        public BarcodeDecoder(IOptions<BarcodeConfig> optionsAccessor)
+        {
+            _config = optionsAccessor.Value;
+
+            barcodeReader = new ZXing.BarcodeReader
+            {
+                AutoRotate = true,
+                TryInverted = true,
+                Options = new DecodingOptions
+                {
+                    TryHarder = true,
+                    PossibleFormats = _config.PossibleFormats
+                }
+            };
+            results = new List<DecodeResult>();
+        }
+
+        public void StartDecoding()
+        {
+            _timeStart = DateTime.Now.Ticks;
+            if (!File.Exists(_config.FilePath))
+                throw new Exception("File not found!");
+
+            // initialize with list of images based on format
```

Comment:
This part of code that loads Bitmaps either from TIFF multipage or single TIF must be replaced, as previously discussed, with an **interface** (with a ReadBitmaps method) that has an implementation that contains the code from here (TiffBitmapReader - reads multipage or single page tiffs).

Reply:
Added requested changes